### ISSUE TITLE
Syncthing: Support DS116 (Armada385)

### ISSUE THREAD

**Issue by MrTango** [SynoCommunity/spksrc#2628]
_For new Package Requests, see the guidelines_  

### Setup
_Package Name:_ syncthing
_Package Version:_ 

_NAS Model:_ DS116
_NAS Architecture:_ Armada385
_DSM version:_ DSM 6.0.2-8451 Update 9

### Expected behavior
It would be really nice to have syncthing also on the DS116, but so far it is not available there.
I have it installed on my DS213j and would like to sync both devices and my laptop with syncthing.

### Actual behavior
Syncthing is not available at the DS116.

### Steps to reproduce
_1._ Add the community repro
_2._ Enable Beta software
_3._ find no syncthing in the package list

**Comment by o0v7r0o** [comment]
I may be wrong, but it is already supported, probably just the build on the server is missing somehow. At least I can compile it for armada38x arch. Hopefully when #2553 will be merged, you will see the spk.

**Comment by Dr-Bean** [comment]
@o0v7r0o @MrTango While we can publish the package (it'll happen semi-automatically with a new release), it won't help in this case: Syncthing isn't yet DSM6 compatible. While the package will show up in Package Center once published, it won't start. See #2216: we're working on it.

**Comment by o0v7r0o** [comment]
@Dr-Bean actually I'm on 6.1beta now and have no problems with packages (both official and community). I jumped out from DSM6 because they broken hibernation somehow.

**Comment by Dr-Bean** [comment]
Eh? You can install any SynoCommunity package (clean! not upgrading from a previously installed package, no manual changes), and it creates a user and the package runs? Cause that's the main issue here...

**Comment by o0v7r0o** [comment]
@Dr-Bean it seems so, as soon the package has its own busybox with adduser. Later today I will check it through.

**Comment by Dr-Bean** [comment]
@o0v7r0o Assuming it's a fluke in the DSM6.1 beta, and in case you're interested what adding DSM6 support looks like for our packages, see https://github.com/SynoCommunity/spksrc/commit/f4ffb15c6949b8464956b5c3736290f97a1c177f.
Although the Syncthing-Inotify package wouldn't need the privilege file, and some other things can be skipped for that package as well, seeing as it uses the user managed via Syncthing.

**Comment by o0v7r0o** [comment]
@Dr-Bean, I've tried to install icecast:
`root@---:~# cat /etc/passwd | grep -i icecast`
`root@---:~# cat /etc/passwd | grep -i icecast`
`icecast:x:101:100:icecast User:/usr/local/icecast/var:/bin/sh`
`root@---:~# /var/packages/icecast/scripts/start-stop-status status`
`icecast is not running`
`root@---:~# /var/packages/icecast/scripts/start-stop-status start`
`Starting icecast ...`
`root@---:~# /var/packages/icecast/scripts/start-stop-status status`
`icecast is running`
`root@---:~# ps ax | grep -i icec`
`19081 ?        Sl     0:00 /usr/local/icecast/bin/icecast -c /usr/local/icecast/var/icecast.xml`
`19091 pts/1    S+     0:00 grep --color=auto -i icec`
`root@---:~# /var/packages/icecast/scripts/start-stop-status stop`
`Stopping icecast ...`
`root@---:~# /var/packages/icecast/scripts/start-stop-status status`
`icecast is not running`
`root@---:~# ps ax | grep -i icec`
`19115 pts/1    S+     0:00 grep --color=auto -i icec`

**Comment by Dr-Bean** [comment]
That's surprising :) What are the chances that'll hold up when DSM6.1 final is released, I wonder ;)

I do think we'll still be going ahead with the approach in the `dsm6` branch, because unless Synology provides official support for users created with busybox, it's subject to change at any point in time: can't have that...

**Comment by Dr-Bean** [comment]
Package for Armada385 has been published, although not with DSM6 support yet. See my previous comment for that.
I'll merge master to the dsm6 branch when possible (as I'm working on other stuff, so I can't switch), but once done, people are able to build a DSM6-compatible package for Syncthing and install locally. Otherwise, as mentioned, subscribe to #2216 to keep informed of progress.

**Comment by smiley-avc** [comment]
@Dr-Bean I've noticed, that you merged master to dsm6 branch. What's the status of DSM6.x support (and more specifically Syncthing on 6.x)? Is it usable? I'm planning to use it on my 416j (armada38x) with 6.x version and I'm a little worried this would break something in the system. I don't want to reinstall the system and thus if something went wrong, I would have to attempt manually fix/remove Syncthing package.

**Comment by Dr-Bean** [comment]
@smiley-avc The `dsm6` branch should generally be on the right track to produce dsm6-compatible packages, although it's not complete yet for a number of packages. For Syncthing, I would think it's more or less done: while I have not tested that specific package, the approach is generally similar to a lot of other packages.
If you build that package, it'll either produce a package that works exactly as intended, or it won't start, but there's no risk of bricking your system. So, if you want to compile it yourself and install manually via Package Center, go right ahead. Please provide feedback if you do :)

**Comment by Dr-Bean** [comment]
One thing I just realized is that Syncthing doesn't have the wizard yet (it tells you something like: use the `sc-syncthing` group to set permissions in DSM, similar to this line: https://github.com/SynoCommunity/spksrc/blob/dsm6/spk/couchpotatoserver-custom/src/wizard/install_uifile_enu#L23-L27

**Comment by smiley-avc** [comment]
@Dr-Bean: Thanks for info. Apologies for late feedback, I've already tried this a week ago, but didn't have the time to get back to you.

I've created that wizard file (containing just empty JSON array) and built the package.

The package correctly installed on my DS416j (with DSM 6.0.2-8451 Update 9) and it was possible start/stop Syncthing via Package Center without any problems.

However, I did have some problems getting Syncthing connected with the one running on my PC. Default discovery of other devices via local announce did not work at all. Naturally, I've enabled firewall rule for Syncthing, but that didn't work. I tried using default port (21027), random one, disabling/enabling firewall on DS, on PC. Nothing.

The only thing that did work, was when I've connected directly from my PC to DS via IP and port (i.e. I've set IP and port of Syncthing, that's running on DS, on Syncthing running on my PC). After that, everything worked as expected. I use schedule to power on/off DS and Syncthing starts up just fine.

The package works as expected, so I guess not much remaining on your part here (except that wizard and other polishing). I'll try to find some time and report this to Syncthing's developer.

Thanks a lot!
